opm/simulators/utils/PartiallySupportedKeywordOptions.hpp

```diff
@@ -0,0 +1,104 @@
+/*
+  Copyright 2021 Equinor.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <map>
+#include <opm/parser/eclipse/Parser/ErrorGuard.hpp>
+#include <opm/parser/eclipse/Parser/ParseContext.hpp>
+#include <opm/simulators/flow/KeywordValidation.hpp>
+
+namespace Opm::KeywordValidation
+{
+
+
+PartiallySupportedOptions<std::string> partially_supported_string_options = {
+    {
+        "COMPORD",
+        {
+            {
+                "ORDER_TYPE",
+                {
+                    .critical = false,
+                    .permitted_values = {"INPUT"},
+                    .message = std::nullopt,
```

Comment:
I'm sorry, but you can't use this.  We don't have C++20 and therefore cannot use designated initialisers.

Reply:
Ah, pity,, then we will use list-initialization for now. Can see later if we need something more friendly.